### DISCUSSION TITLE
Resources on Custom Modules

### DISCUSSION THREAD

**Discussion by NecoHorne** [directus/directus#13061] [Archive]
Hi is there any beter resources / documentation on how to create custom modules? 

The documentation currently available does not explain how to use the sidebar, how the navigation works etc. 

from the documentation:
```
A route component provides an blank canvas for creating anything you need. You can use the globally registered private-view component to get access to Directus' page structure consisting of the module bar, the navigation, the sidebar, the header and the main content area.
```

But it gives no example or documentation on this object ie how to access and use the sidebar or navigation. I went to the readme here : [private view readme](https://github.com/directus/directus/blob/3445114296cb66a0ac5d9ddbf8c649bff11a3e5b/app/src/views/private/readme.md) but its blank. 

Following the documentation [extensions/modules/](https://docs.directus.io/extensions/modules/) I can setup the module, build it an include it in the project, the navigation is just a bit of an unknown at the moment. 

![image](https://user-images.githubusercontent.com/30078491/165939189-3140499c-6684-43b8-ac34-cd818866cc4c.png)

Currently on V9.9.1

**Comment by rijkvanzanten** [comment]
There's currently no written in-depth docs on how to utilize the `private-view` component. The source code is currently your best bet (look for `<slot name="x">` https://github.com/directus/directus/blob/e47502e11fc173f7cb9524fc6e853de548dd8389/app/src/views/private/private-view.vue
